File: jambeth/jambeth-merge/src/main/java/de/osthus/ambeth/merge/incremental/IIncrementalMergeState.java
package de.osthus.ambeth.merge.incremental;

import de.osthus.ambeth.cache.ICache;
import de.osthus.ambeth.merge.model.CreateOrUpdateContainerBuild;

public interface IIncrementalMergeState
{
	ICache getStateCache();

	CreateOrUpdateContainerBuild newCreateContainer(Class<?> entityType);

	CreateOrUpdateContainerBuild newUpdateContainer(Class<?> entityType);
}

File: ambeth/Ambeth.Merge/ambeth/merge/incremental/IIncrementalMergeState.cs
using De.Osthus.Ambeth.Cache;
using De.Osthus.Ambeth.Merge.Model;
using System;

namespace De.Osthus.Ambeth.Merge.Incremental
{
    public interface IIncrementalMergeState
    {
        ICache GetStateCache();

        CreateOrUpdateContainerBuild NewCreateContainer(Type entityType);

        CreateOrUpdateContainerBuild NewUpdateContainer(Type entityType);
    }
}
